naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/schema/NaptimeUnionField.scala

```diff
@@ -80,9 +82,22 @@ object NaptimeUnionField {
         subTypeField.fieldType,
         resolve = context => {
           if (unionDataSchema.getProperties.containsKey(TYPED_DEFINITION_KEY)) {
-            Option(context.value.element.getDataMap("definition"))
+            val resultAsDataMap: Option[Action[SangriaGraphQlContext, Any]] = Try(
+              context.value.element
+                .getDataMap("definition")).toOption
               .map(element => context.value.copy(element = element))
-              .getOrElse(subTypeField.resolve(context))
+            resultAsDataMap.getOrElse {
+              // Create a separate field to access the non data-map directly. This field will be
+              // found at the "definition" key.
```

Comment:
let's add the `why` for this comment, I think the previous impl assumed each union member is a data map which may not be true? e.g a union where one member is a string or enum may make sense?

Reply:
I've simplified the code and updated the documentation. Let me know if it makes more sense.